### ISSUE TITLE
Add gRPC API to core Mainflux services

### ISSUE THREAD

**Issue by anovakovic01** [absmach/magistrala#432]
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.
No.

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
We should add `gRPC` API to core Mainflux services. Benefit will be great as we will give support for single API over two protocols `HTTP` and `gRPC`. Also, our SDKs will communicate over gRPC which will have much better performance.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
IMO we should have this feature.

**Comment by drasko** [comment]
We should also consider using [gRPC Gateway](https://github.com/grpc-ecosystem/grpc-gateway), and providing HTTP REST API that way, rather than keeping current implementation once gRPC is in place.
